src/State/Observer.lua

```diff
@@ -39,8 +39,11 @@ end
 	will be held in memory, preventing GC, so disconnecting is important.
 ]]
 function class:onChange(callback: () -> ()): () -> ()
+	local uniqueIdentifier = tostring(self._createdChangeListeners)
+
+	self._createdChangeListeners += 1
```

Comment:
What's the difference between `_createdChangeListeners` and `_numChangeListeners` here? Also, why `tostring` it?

Reply:
``_numChangeListeners`` decrements whenever a listener is removed, whereas ``_createdChangeListeners`` constantly increases with each additional listener (and never decreases until there's 0 listeners connected). This is intentional so that we can have a unique identifier for each listener connection that doesn't overwrite any previous connections.

The ``tostring`` is so that it gets stored as a dictionary from the start instead of an array. That being said, since we aren't using ``table.remove()`` this seems a bit redundant and unnecessary.

Reply:
Fair enough. I'd normally do `local uniqueIdentifier = {}` since this is stateless and always saved as a dictionary key.

Reply:
That's interesting, I never thought of doing it that way, but I like it. Thanks for the feedback! I'll work on changing that up.

Reply:
I made some changes based on your suggestion. I also added some unit testing to ``Observer.lua`` to verify that everything in Observer still works as intended.